---
title: "**Introduction to RNAseq workshop: reads to differential gene expression**"
output:
  html_document:
    toc: false
    toc_float: false
from: markdown+emoji
---

This workshop series will introduce you to RNA sequencing data analysis for differential expression and functional enrichment on the command line. We will use the nf-core/rnaseq pipeline to process raw sequence reads to a gene count matrix and then perform differential expression and enrichment analyses with R/Rstudio.

## Target audience{.unlisted}

This workshop series is suitable for people who are familiar with working at command line interface and may be new to RNAseq for differential expression analysis. The course is beginner friendly and intended for those interested in using the command line interface for their analysis. It is also suitable for those who want to learn about and use nf-co.re workflows.

## Prerequisites{.unlisted}

* [Familiarity with Unix/Linux command line](https://linuxjourney.com/lesson/the-shell)
* [Familiarity with R/RStudio](https://swcarpentry.github.io/r-novice-gapminder/index.html)
* [Introduction to RNAseq webinar](https://www.youtube.com/watch?v=tITR3WR_jWI) ([slides](https://doi.org/10.5281/zenodo.8323207))

## Get the most out of this workshop{.unlisted}

This workshop series is presented across 2 days and will consist of code-along and breakout room sessions. You will get the most out of the workshop if you:

* Brush up on your command line basics before Day 1
* Brush up on your R/Rstudio skills
* Watch our webinar introducing RNAseq experiments 
* Set up your computer before Day 1
* Use the Slack channel to ask questions and engage in discussions during the workshop
* Do the provided exercises during the workshop

## Workshop schedule{.unlisted}

|Lesson               |Overview                                                                                                                                                                                    |
|---------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|[Set up](https://sydney-informatics-hub.github.io/rnaseq-workshop-2023/setup.html) | Follow these instructions to install a terminal application and web browser to be used in the workshop.                                                                                     |
|[Day 1](https://sydney-informatics-hub.github.io/rnaseq-workshop-2023/notebooks/1.1_rnaseq-experiment.html)                | Participants will use nf-core’s rnaseq pipeline to transform raw sequencing data into analysis ready count data.                                               |
|[Day 2](https://sydney-informatics-hub.github.io/rnaseq-workshop-2023/notebooks/2.1_rstudio-downstream.html)                | Participants will use RStudio to perform differential expression and enrichment analysis on count matrix.|

## Course survey!{.unlisted}

Please fill out our [course survey](https://www.surveymonkey.com/r/rnaseq2023) before you leave! Help us help you! :smiley:

## Credits and acknowledgements{.unlisted}

Lead trainers: 

* Dr Nandan Deshpande 
* Dr Georgie Samaha 

This training was developed under the Australian BioCommons ‘Bring Your Own Data’ - Command Line Interface (CLI) Platform Expansion Project by the Sydney Informatics Hub, in partnership with Pawsey Supercomputing Centre and Australian Academic Research Network (AARNet). This project aims to create highly accessible, highly available, highly scalable analysis and data sharing capabilities for the benefit of life science researchers nationally.

![](./Figs/0.0_logos.png)