---
title: "**Workshop summary**"
output:
  html_document:
    toc: false
    toc_float: false
from: markdown+emoji
---

## 2.6.1 Day 1: Raw sequence to gene counts**
On Day 1 we used the nf-core/rnaseq pipeline to convert raw RNAseq data into raw counts.
<img src="../Figs/1.1_nfcore-rnaseq.png" alt="Your Alt Text" style="width: 70%;">

We discussed the following essential steps in a RNAseq analysis workflow: 
    - Check the quality of raw-reads.
    - Trim the raw-reads to get rid of bad-quality read-regions and/or bad-quality reads.
    - Align the trimmed-reads to reference sequence to identify where they belong.
    - Quantify the aligned reads to get gene-level read-counts.
- We discussed the workflow management tool nextflow which can be used for automated, reproducible, flexible and portable analysis.
- We excecuted the nf-core/rnaseq pipeline and interpreted its outputs.


## 2.6.2 Day 2: Counts to genes and functional enrichments**
On Day 2 we used the raw counts to identify the differentially expressed genes and highlighted the relevant functions.
<img src="../Figs/2.1_Rstudio_chrome_rmd.png" alt="Your Alt Text" style="width: 70%;">
The following steps were identified to be essential :
  - Perform an exploratory analysis of the count data for quality control.
  - Analyse the count data to identify differentially expressed genes.
  - Identify functional enrichments from differentially expressed genes.
- We also discussed how to perform reproducible analysis using Rstudio with Rmarkdown using singularity containers. 